galsim/wfirst/wfirst_psfs.py

```diff
@@ -0,0 +1,460 @@
+                               including extensions for any compression that is to be done.  See
+                               galsim.fits.write documentation for information about compression
+                               options.
+    @param data_filename       The name of the file to which information about the PSF images is to
+                               be written, for use by the getStoredPSF() routine.
+    @param bandpass_list       A list of bandpasses for which images should be generated and
+                               stored.  If None, all WFIRST imaging passbands are used.
+                               [default: None]
+    @param clobber             Should the routine clobber `PSF_list` and `image_filename` (if they
+                               already exist? [default: False]
+    """
+    # Check for sane input PSF_list.
+    if len(PSF_list) != galsim.wfirst.n_sca+1:
+        raise ValueError("PSF_list must have length %d, as output by getPSF!"
+                         %(galsim.wfirst.n_sca+1))
+
+    # Check if file already exists and warn about clobbering.
+    if os.path.exists(image_filename) or os.path.exists(data_filename):
+        if clobber is False:
+            raise ValueError("At least one output file already exists, and clobber is not set!")
+        else:
+            import warnings
+            warnings.warn("At least one output file already exists, and will be clobbered.")
+
+    # Check that bandpass list input is okay.  It should be strictly a subset of the default list of
+    # bandpasses.
+    if bandpass_list is None:
+        bandpass_list = default_bandpass_list
+    else:
+        if not set(bandpass_list).issubset(default_bandpass_list):
+            err_msg = ''
+            for item in default_bandpass_list:
+                err_msg += item+' '
+            raise ValueError("Bandpass list must be a subset of the default list, containing %s"
+                             %err_msg)
+
+    # Get all the WFIRST bandpasses.
+    bandpass_dict = galsim.wfirst.getBandpasses()
+
+    # Loop through making images and lists of their relevant parameters.
+    im_list = []
+    bp_name_list = []
+    SCA_index_list = []
+    for bp_name in bandpass_list:
+        bandpass = bandpass_dict[bp_name]
+
+        for SCA in range(galsim.wfirst.n_sca+1):
+            if PSF_list[SCA] is None:
+                continue
+
+            PSF = PSF_list[SCA]
+            if not isinstance(PSF, galsim.ChromaticOpticalPSF):
+                raise RuntimeError("Error, PSFs are not ChromaticOpticalPSFs.")
+            im = PSF.drawImage(bandpass, scale=0.5*galsim.wfirst.pixel_scale)
+            im_list.append(im)
+            bp_name_list.append(bp_name)
+            SCA_index_list.append(SCA)
+
+    # Save images to file.
+    n_ims = len(im_list)
+    galsim.fits.writeMulti(im_list, image_filename, clobber=clobber)
+    print 'Saved %d images to file %s'%(n_ims, image_filename)
```

Comment:
Shouldn't have any print statements.  Although fine if you want to accept a logger parameter as I suggested above.